lib/ProductOpener/Display.pm

```diff
@@ -1069,6 +1094,60 @@ sub display_no_index_page_and_exit () {
 	exit();
 }
 
+=head2 display_robots_txt_and_exit ($request_ref)
+
+Return robots.txt page and exit.
+
+robots.txt is dynamically generated based on lc, it's content depends on $request_ref:
+- if $request_ref->{deny_all_robots_txt} is 1: a robots.txt where we deny all traffic
+  combinations.
+- otherwise: the standard robots.txt. We disallow indexing of most facet pages, the
+  exceptions can be found in ProductOpener::Config::index_tag_types
+
+=cut
+
+sub display_robots_txt_and_exit ($request_ref) {
+	my $template_data_ref = {facets => []};
+	my $vars = {deny_access => $request_ref->{deny_all_robots_txt}, disallow_paths_localized => []};
+
+	foreach my $type (keys %tag_type_singular) {
+		# Get facet name for both english and the request language
+		foreach my $lang ('en', $request_ref->{lc}) {
+			my $tag_value_singular = $tag_type_singular{$type}{$lang};
+			my $tag_value_plural = $tag_type_plural{$type}{$lang};
+			if (
+					defined $tag_value_singular
+				and length($tag_value_singular) != 0
+				and not($tag_value_singular ~~ $vars->{disallow_paths_localized})
+				# check that it's not one of the exception
+				# we don't perform this check below for list of tags pages as all list of
+				# tags pages are not indexable
+				and not($type ~~ @ProductOpener::Config::index_tag_types)
```

Comment:
in the initialization of Display.pm, we could create a %index_tag_types hash with keys for all entries in @index_tag_types so that we can directly check if the key exists, instead of iterating on each array entry.

On a side note, the ~~ smartmatch operator is deprecated in the newest Perl and will be removed in future versions.